components/properties/src/script.rs

```diff
@@ -109,16 +191,67 @@ pub struct ScriptExtensions<'data> {
 }
 
 impl<'data> ScriptExtensions<'data> {
-    pub fn try_new(
+    // This method is intended to be used by constructors of deserialized data
+    // in a data provider.
+    #[doc(hidden)]
+    pub fn new(
         trie: CodePointTrie<'data, ScriptWithExt>,
         extensions: VarZeroVec<'data, ZeroSlice<Script>>,
-    ) -> Result<ScriptExtensions<'data>, PropertiesError> {
-        // TODO: do validation here
-
-        Ok(ScriptExtensions { trie, extensions })
+    ) -> ScriptExtensions<'data> {
+        ScriptExtensions { trie, extensions }
     }
 
     /// Returns the `Script` property value for this code point.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_provider::prelude::*;
+    /// use icu_properties::provider::key;
+    /// use icu::properties::provider::ScriptExtensionsPropertyV1Marker;
+    /// use icu_properties::Script;
+    /// use icu::properties::script::ScriptExtensions;
+    ///
+    /// let provider = icu_testdata::get_provider();
+    ///
+    /// let payload: DataPayload<ScriptExtensionsPropertyV1Marker> = provider
+    ///     .load_payload(
+    ///         key::SCRIPT_EXTENSIONS_V1,
+    ///         &DataRequest {
+    ///             options: ResourceOptions::default(),
+    ///             metadata: Default::default(),
+    ///         },
+    ///     )
+    ///     .expect("The data should be valid")
+    ///     .take_payload()
+    ///     .expect("Loading was successful");
+    ///
+    /// let scx: &ScriptExtensions = &payload.get().data;
```

Comment:
I find it very confusing that the type is called ScriptExtensions, and the sample variable name scx, but it calls .get_script_val(cp) and the docs do say "Returns the `Script` property value". scx should return scx values not sc values.

The variable is just in the sample code and could be something like sc_and_scx.

However, is the name `icu::properties::script::ScriptExtensions` right?
If the type provides access to both the Script property and the ScriptExtensions property, then naming it for just one of the two will be confusing.

Reply:
Hmm, yes, the names do seem misleading -- in other words, I wouldn't blame a user at all for being confused by this.

We named the newtype-style enum for the trie values here as `ScriptWithExt`.  Would renaming the containing struct in question from `ScriptExtensions` to `ScriptWithExtensions` be reasonable?

Reply:
ScriptWithExtensions would probably be better, because it's not the name of either property.

What about making this an internal type name, and exposing Script vs. ScriptExtensions (or ScriptProperty vs. ScriptExtensionsProperty or similar) as public API, with each type having functions specific to its property (eg either get_script_val() or has_script() but not both), and calling into the shared implementation?

Reply:
`ScriptWithExtensions` SGTM.

I prefer a single shared type because it is a 1:1 correspondence with the data payload type.

Reply:
Done. I updated the name of the struct and other things (ex: providers, data markers) with derivative names, and the local variable names in examples / tests.

I left the structure of the public-facing struct as it is.  It's consistent with what we do for CPT / UnicodeSet, and IIRC, doing so allows the user to manage the lifecycle of the data however they want. (And I think we've had discussions a few times before sketching out how we might solve the problem of making this more user-friendly.)